_posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md

```diff
@@ -0,0 +1,256 @@
+
+Data Prepper now supports routing traces and logs to different indexes dynamically. To achieve this, you can define an 
+index name using a format string which can include properties from different events. With this capability, Data Prepper 
+can support an arbitrary number of indexes using a single sink. For example, if you have logs with different 
+application identifiers, you can route events to an index specifically for each application.
+
+Data Prepper also now also allows routing documents to specific OpenSearch shards using the existing `routing` parameter 
+in OpenSearch. The new `routing_field` property on the OpenSearch sink will use properties from events to specify how to 
+route within OpenSearch. Most users prefer allowing OpenSearch to generate document IDs and choose the shard 
+from that. However, Data Prepper now allows you to specify these values explicitly.
+
+Going hand-in-hand with the routing field, Data Prepper now lets you specify a complex field to 
+use when specifying a document ID. Previously, you could only copy a value to a field in the root of 
+the event to use it as a document ID.
+
+## Type conversion
+
+Data Prepper now supports a new 
+[convert entry](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/mutate-event-processors#convertentryprocessor) 
+processor as part of 
+[mutate processors](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/mutate-event-processors#mutate-event-processors) 
+to convert a value from one type to another. This can be particularly useful in conditional expressions when using conditional routing.
+
+For example, using the following processor in a pipeline lets you convert the value of `status` key to `integer`: 
+
+```
+...
+processor:
+  - convert_entry_type:
+      key: "status"
+      type: "integer"
+...
+```
+
+## Other improvements
+
+In addition to the new features already described, Data Prepper 2.1.0 has several other important improvements:
+
+* Data Prepper pipelines with OpenSearch sinks now wait for OpenSearch to start during pipeline initialization if the sink is not available instead of failing.
+*  OpenSearch sink now allows you to load index mapping and ISM template files from an S3 bucket. You can configure the S3 object by providing its S3 URI in existing configuration option to load them.
+* Data Prepper expressions now support the `null` value keyword in conditions with equality operators.
+* To help with pipeline stability, Data Prepper now supports a circuit breaker on Java heap usage. The circuit breaker will cause the sources to stop accepting new data that could possibly cause the process to crash.
+* The documentation team has been updating the [Data Prepper documentation](https://opensearch.org/docs/latest/data-prepper/index/) by migrating many of the details from the GitHub repository into the primary documentation. 
```

Comment:
```suggestion
* The Documentation team has been updating the [Data Prepper documentation](https://opensearch.org/docs/latest/data-prepper/index/) by migrating many of the details from the GitHub repository to the primary documentation. 
```

Reply:
Is this sentence necessary? We should probably avoid referring to the Documentation team as a discrete entity, given that this is open source, and I'm not sure how much value this sentence adds.

Reply:
We can remove it. I was calling it out because the documentation is much better than it used to be.